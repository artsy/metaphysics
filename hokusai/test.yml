version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
