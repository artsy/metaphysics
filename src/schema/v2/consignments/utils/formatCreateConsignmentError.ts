import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
