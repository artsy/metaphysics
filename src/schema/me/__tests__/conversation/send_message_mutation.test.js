import { runAuthenticatedQuery } from "test/utils"

describe("SendConversationMessageMutation", () => {
  it("creates a message and returns its new data payload", () => {
    const mutation = `
      mutation {
        sendConversationMessage(
          input: {
            id: "623",
            from: "pio-dog@example.com",
            body_text: "Sehr schÃ¶n!"
            reply_to_message_id: "221"
          }
        ) {
            conversation {
              id
            }
            messageEdge {
              cursor
              node {
                id
                body
                from {
                  email
                  name
                }
              }
            }
          }
      }
    `

    const rootValue = {
      conversationLoader: () =>
        Promise.resolve({
          id: "420",
          initial_message: "10/10 would buy",
          from_email: "gallery@example.com",
          to: ["1234567"],
          to_name: "Some Gallery",
        }),
      conversationCreateMessageLoader: () =>
        Promise.resolve({
          id: "222",
          radiation_message_id: "333",
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(
      ({ sendConversationMessage }) => {
        expect(sendConversationMessage).toMatchSnapshot()
      }
    )
  })
})
