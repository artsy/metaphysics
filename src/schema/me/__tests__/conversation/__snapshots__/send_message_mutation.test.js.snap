// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendConversationMessageMutation creates a message and returns its new data payload 1`] = `
Object {
  "conversation": Object {
    "id": "420",
  },
  "messageEdge": Object {
    "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
    "node": Object {
      "body": "Sehr sch√∂n!",
      "from": Object {
        "email": "pio-dog@example.com",
        "name": null,
      },
      "id": "222",
    },
  },
}
`;
